### PR TITLE
Added new module: bbmap/sendsketch

### PR DESCRIPTION
**Pull request by phanhung2**
This is a module for the BBMap tool `sendsketch.sh`

## PR checklist

Closes #XXX 

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] If necessary, include test data in your PR.
- [X] Remove all TODO statements.
- [X] Emit the `versions.yml` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [X] Add a resource `label`
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [X] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [X] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [X] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/bbmap/sendsketch/main.nf [hunk 1]
```diff
@@ -0,0 +1,35 @@
+process BBMAP_SENDSKETCH {
+    tag "$file"
```
Comment (nvnieuwk) [inline modules/nf-core/bbmap/sendsketch/main.nf:2]:
```suggestion
    tag "$meta.id"
```

This should use `meta.id` if meta is an input

File: modules/nf-core/bbmap/sendsketch/main.nf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    output:
+    tuple val(meta), path('results.txt')  , emit: hits
+    path "versions.yml"                   , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
```
Comment (nvnieuwk) [inline modules/nf-core/bbmap/sendsketch/main.nf:21]:
```suggestion
    def args = task.ext.args ?: ''
    def prefix =  task.ext.prefix ?: "${meta.id}"
```

File: modules/nf-core/bbmap/sendsketch/main.nf [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    def file_used = file.size() > 1 ? file[0] : file
+    """    
+    sendsketch.sh \\
+        in=${file_used} \\
+        out='results.txt' \\
```
Comment (nvnieuwk) [inline modules/nf-core/bbmap/sendsketch/main.nf:26]:
```suggestion
        out=${prefix}.txt \\
```

File: modules/nf-core/bbmap/sendsketch/main.nf [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    conda "bioconda::bbmap=39.01"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bbmap:39.01--h5c4e2a8_0':
+        'quay.io/biocontainers/bbmap:39.01--h5c4e2a8_0' }"
+
+    input:
+    tuple val(meta), path(file)
+
+    output:
+    tuple val(meta), path('results.txt')  , emit: hits
```
Comment (nvnieuwk) [inline modules/nf-core/bbmap/sendsketch/main.nf:14]:
```suggestion
    tuple val(meta), path("*.txt")  , emit: hits
```

File: modules/nf-core/bbmap/sendsketch/main.nf [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+    script:
+    def args = task.ext.args ?: ''
+    def file_used = file.size() > 1 ? file[0] : file
+    """    
+    sendsketch.sh \\
+        in=${file_used} \\
+        out='results.txt' \\
+        $args \\
+        -Xmx${task.memory.toGiga()}g
```
Comment (nvnieuwk) [inline modules/nf-core/bbmap/sendsketch/main.nf:28]: It's better to add a check and see if the memory has been set (see this module for an example: https://github.com/nf-core/modules/blob/master/modules/nf-core/gatk4/combinegvcfs/main.nf)

File: modules/nf-core/bbmap/sendsketch/meta.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - versions:
+      type: file
+      description: File containing software versions
+      pattern: "versions.yml"
+  - results.txt:
```
Comment (nvnieuwk) [inline modules/nf-core/bbmap/sendsketch/meta.yml:40]:
```suggestion
  - hits:
```

**Review by nvnieuwk — CHANGES_REQUESTED** [review]

**Comment by phanhung2** [comment]
Thank you @nvnieuwk! These changes have been very helpful and have been implemented to the module. If you could take a review and let me know, it would be greatly appreciated!

**Review by nvnieuwk — APPROVED** [review]
LGTM!

**Pull request merged**